Resources/public/js/views/services/ez-contenteditviewservice.js

```diff
@@ -60,9 +65,123 @@ YUI.add('ez-contenteditviewservice', function (Y) {
                 service._loadContentType(resources.ContentType, tasks.add());
 
                 tasks.done(function () {
-                    next(service);
+                    service._setVersionFields(content, translationExists, function () {
+                        next(service);
+                    });
+                });
+            });
+        },
+
+        /**
+         * Checks if given languageCode is included in translations list of given content
+         *
+         * @method _checkIfTranslationExists
+         * @private
+         * @param {Y.eZ.Content} content
+         * @param {String} languageCode
+         * @return {Boolean}
+         */
+        _checkIfTranslationExists: function (content, languageCode) {
+            var translationExists;
+
+            translationExists = Y.Array.find(
+                content.get('currentVersion').getTranslationsList(),
+                function (translation) {
+                    return (translation === languageCode);
+                }
+            );
+
+            return !!translationExists;
+        },
+
+        /**
+         * Sets fields of edited version
+         *
+         * @method _setVersionFields
+         * @private
+         * @param {Y.eZ.Content} content
+         * @param {Boolean} translationExists defines if fields will be set for existing translation
+         * @param {Function} callback
+         */
+        _setVersionFields: function (content, translationExists, callback) {
+            var fields,
+                version = this.get('version');
+
+            fields = this._getFieldsForEdit(content, translationExists);
+
+            version.set('fields', fields);
+            callback();
+        },
+
+        /**
+         * Returns object containing fields definitions for given content.
+         * If editing content in context of creating new translation it returns fields based on default field
+         * defitnitions of ContentType if no baseLanguageCode was set, otherwise if baseLanguageCode was set
+         * it returns fields cloned from loaded content with setting proper languageCode.
+         * If editing content in context of editing existing translation it returns fields from loaded content
+         * which is default behaviour.
+         *
+         * @method _getFieldsForEdit
+         * @private
+         * @param {Y.eZ.Content} content
+         * @param {Boolean} translationExists
+         * @return {Object}
+         */
+        _getFieldsForEdit: function (content, translationExists) {
+            var baseLanguageCode = this.get('baseLanguageCode'),
+                languageCode = this.get('languageCode'),
+                contentFields = content.get('fields'),
+                setDefaultFields = false,
+                fields;
+
+            if (baseLanguageCode) {
+                if (!translationExists && !content.get('alwaysAvailable')) {
```

Comment:
as I wrote in https://github.com/ezsystems/PlatformUIBundle/pull/280#discussion_r38178912, this check should be the first thing done in the `_loadContent` callback, doing it here is way too late (and you don't have to check the alwaysAvailable flag).